crates/driver/src/logic/competition/order/mod.rs

```diff
@@ -0,0 +1,148 @@
+use {
+    crate::{logic::eth, util},
+    primitive_types::U256,
+};
+
+pub mod signature;
+
+pub use signature::Signature;
+
+/// An order in the auction.
+#[derive(Debug)]
+pub struct Order {
+    pub uid: Uid,
+    pub from: eth::Address,
+    pub receiver: Option<eth::Address>,
+    pub valid_to: util::Timestamp,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: Side,
+    pub fee: eth::Asset,
```

Comment:
I think that the order domain object needs two fees:
- The fee that is signed by the user (used for building settlement calldata)
- The "full fee amount" that is used for settlement competition.

Reply:
Looks great.